lib/standard/loads_yaml_config.rb

```diff
@@ -25,15 +22,17 @@ def load_standard_yaml(yaml_path)
       end
     end
 
-    def construct_config(yaml_path, standard_yaml)
+    def construct_config(yaml_path, standard_yaml, todo_path, todo_yaml)
       {
         ruby_version: Gem::Version.new((standard_yaml["ruby_version"] || RUBY_VERSION)),
         fix: !!standard_yaml["fix"],
         format: standard_yaml["format"],
         parallel: !!standard_yaml["parallel"],
-        ignore: expand_ignore_config(standard_yaml["ignore"]),
+        ignore: expand_ignore_config(standard_yaml["ignore"]) + expand_ignore_config(todo_yaml["ignore"]),
```

Comment:
What happens if there are duplicates or contradictions between these two sets of ignore rules?

Reply:
Good question and my answer is I don't know.  I'll do some testing to see what currently happens.  Then we can make a decision if the settings in `.standard.yml` should be overridden by the `.standard_todo.yml` settings.

Reply:
@searls what currently happens is the ignore list from the `.standard.yml` and `.standard_todo.yml` are combined into one list with the possibility of duplicates.  Duplicates are handled without any issue.

Actually the patterns in the ignore section for both files is translated into a Rubocop AllCops Exclude section, duplicates and all.  So really Rubocop has not problem with duplicate ignores and uses the broadest ignore pattern.

In the case where one has a broader ignore pattern then the other then the broader one wins.  For example the below will ignore all the files in `foo/bar`.  It does not matter which file the `foo/bar/**/*` is in.

```
# .standard.yml
ignore:
  - foo/bar/**/*
```

```
# .standard_todo.yml
ignore:
  - foo/bar/somefile.rb
```

I don't think we need to add more logic to detect duplicates or overlap with the files.  At least not right now.  If it becomes a problem we can add checks.

My vision for using Standard with the Todo file is as follows:

1) User has an existing project.
2) They install Standard, setup the initial `.standard.yml` config including ignoring folders such as `/tmp`.
3) They run Standard and see all the errors they can't fix right now.
4) They use Standard to generate the Todo file.  Now they can add Standard to their automated build with any new Standard errors caught by the automated build.
5) They remove files from the Todo file as they work on the application.  The warning message is continually printed to remind them about the Todo file.

What are your thoughts?

Reply:
Thanks for this analysis!